### PR TITLE
Explicitly call toString() on values in maps passed to Docker API

### LINKED ISSUE #404: Many properties fail if GString is used instead of String
Example: 
```

ext { 
  imageName =  getProp("imageName", "${group}/" + getProp("image", project.name))
   tagVersion = getProp("tagVersion", "${->version }" )

// ... 
}


task buildImage( type: DockerBuildImage ) {
    // ...
    tags = [ project.imageName , project.tagVersion ]
}

```
Fails with 

 java.lang.ClassCastException: org.codehaus.groovy.runtime.GStringImpl cannot be cast to java.lang.String org.codehaus.groovy.runtime.GStringImpl cannot be cast to java.lang.String
Created null. image: null 

Workaround:

  tags = [ ''+ project.imageName , '' + project.tagVersion ]

### PR DESCRIPTION
**Pull request by orzeh**
Fix #404

### CODE REVIEW

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy [hunk 1]
```diff
@@ -374,7 +374,7 @@ class DockerCreateContainer extends AbstractDockerRemoteApiTask {
         }
 
         if(getLabels()) {
-            containerCommand.withLabels(getLabels())
+            containerCommand.withLabels(getLabels().collectEntries { [it.key, it.value.toString()] })
```
Comment (cdancy) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: Should we do a `toString()` on the key as well?

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy [hunk 2]
```diff
@@ -139,7 +139,7 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         }
 
         if (getLabels()) {
-            buildImageCmd.withLabels(getLabels())
+            buildImageCmd.withLabels(getLabels().collectEntries { [it.key, it.value.toString()] })
```
Comment (cdancy) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy]: Maybe a `toString()` on the key here as well?

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerListImages.groovy [hunk 3]
```diff
@@ -53,7 +53,7 @@ class DockerListImages extends AbstractDockerRemoteApiTask {
         }
 
         if (getLabels()) {
-            listImagesCmd.withLabelFilter(getLabels())
+            listImagesCmd.withLabelFilter(getLabels().collectEntries { [it.key, it.value.toString()] })
```
Comment (cdancy) [inline src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerListImages.groovy]: And here if it makes sense?

File: src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy (hunk 1 above)
Comment (orzeh) [reply src/main/groovy/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.groovy]: I'll check this.

**Comment by orzeh** [comment]
@cdancy I've checked keys as GStrings  and there were no errors. I'll leave it as is. You can merge if there are no other comments/issues to address.

**Comment by cdancy** [comment]
@orzeh sounds good. Merging now.

**Pull request merged**